### ISSUE TITLE
Error in parallel.setNumThreads

### ISSUE THREAD

**Issue by jinsooihm** [glotzerlab/freud#283]
**Describe the bug**
`freud.parallel.setNumThreads` does not seem to change the number of threads correctly in the following situation.

**To Reproduce**
```
#script1.py
import freud
import numpy as np
import time

def main():
    rcut = 1.0
    L = 10
    N = 20000

    fbox = freud.box.Box.cube(L)
    seed = 0
    np.random.seed(seed)
    points = np.random.uniform(-L/2, L/2, (N, 3))

    freud.parallel.setNumThreads(0)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(5)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(3)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(1)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

if __name__=="__main__":
    main()
```

```
#script2.py
import freud
import numpy as np
import time

def main():
    rcut = 1.0
    L = 10
    N = 20000

    fbox = freud.box.Box.cube(L)
    seed = 0
    np.random.seed(seed)
    points = np.random.uniform(-L/2, L/2, (N, 3))

    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(5)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(3)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

    freud.parallel.setNumThreads(1)
    start = time.time()
    lc = freud.locality.LinkCell(fbox, rcut)
    lc.compute(fbox, points, points, exclude_ii=True)
    end = time.time()
    print(end - start)

if __name__=="__main__":
    main()
```
`script1.py` and `script2.py` differ only by one line that `script1.py` calls `freud.parallel.setNumThreads(0)` before the first computation of `LinkCell`, but `script2.py` does not.
It seems like that `freud.parallel.setNumThreads` does not actually modify the number of threads in `script2.py`.

**Error output**
Output of `script1.py`
```
0.10687088966369629
0.11796903610229492
0.14867520332336426
0.3646411895751953
```
Output of `script2.py`
```
0.10623693466186523
0.11173605918884277
0.1130838394165039
0.10568714141845703
```
The output of `script1.py` clearly shows the change in the performance from the decrease in the number of threads, while the output of `script2.py` essentially stays same.

**System configuration (please complete the following information):**
 - OS: macOS
 - Version of Python: 3.6.8
 - Version of freud: 1.0

**Additional context**
Realistically, it is not a big issue since there is not much reason to change the number of threads during execution.

**Comment by bdice** [comment]
@vyasr Does #281 resolve this?

**Comment by vyasr** [comment]
No, I don't believe that it does. @jinsoo960 can you confirm that? 

I'm not sure what causes this issue, but I suspect it's some oddity in how the TBB thread manager works. I concur with Jin Soo that fixing this is not a high priority, though.

**Comment by jinsooihm** [comment]
It does not fix the issue.

**Comment by bdice** [comment]
I was trying to run some benchmarks with 1 thread and I couldn't get `setNumThreads` or the context manager to work. The runtimes appear identical with 4 and 1 thread, and it actually uses all 4 cores of my CPU. I'll assign this for the next release.

**Comment by vyasr** [comment]
Fixed by #298
